src/librustc/ty/mod.rs

```diff
@@ -2222,6 +2222,16 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
             || self.sess.cstore.impl_trait_ref(self.global_tcx(), id))
     }
 
+    /// Returns true if the impl is positive and is for a trait which contains
+    /// no items
+    pub fn impl_always_allowed_to_overlap(self, def_id: DefId) -> bool {
```

Comment:
I think we should change this to take two impls, and call it `impls_are_allowed_to_overlap(impl1, impl2)`. For one thing, we always call it in pairs, but for another we probably want to permit overlap so long as the "polarity" of both impls is the same (that is, I see no reason you can't have overlapping negative impls as well, but indeed you do not want overlapping positive and negative impls).